builder/simplify.cc

```diff
@@ -988,13 +1020,14 @@ class simplifier : public node_mutator {
 
   void visit(const clone_buffer* op) override {
     visit_symbol(op->src, true);
-    auto set_refs = set_value_in_scope(references, op->sym, 0);
+    auto set_refs = set_value_in_scope(references, op->sym, reference_info(0, false));
     auto set_bounds = set_value_in_scope(buffer_bounds, op->sym, buffer_bounds[op->src]);
     stmt body = mutate(op->body);
+    auto& ref_info = references[op->sym];
 
     if (!body.defined()) {
       set_result(stmt());
-    } else if (*references[op->sym] == 0) {
+    } else if (!ref_info || ref_info->counter_ == 0) {
```

Comment:
We should assert or just assume this exists. It can't be undefined unless something is really broken.